tests/integration/distribution.go

```diff
@@ -338,8 +340,156 @@ func (s *CCVTestSuite) TestEndBlockRD() {
 	}
 }
 
+// TestSendRewardsToProvider is effectively a unit test for TestSendRewardsToProvider(),
+// but is written as an integration test to avoid excessive mocking.
+func (s *CCVTestSuite) TestSendRewardsToProvider() {
+	testCases := []struct {
+		name           string
+		setup          func(sdk.Context, *consumerkeeper.Keeper, icstestingutils.TestBankKeeper)
+		expError       bool
+		tokenTransfers int
+	}{
+		{
+			name: "successful token transfer",
+			setup: func(ctx sdk.Context, keeper *consumerkeeper.Keeper, bankKeeper icstestingutils.TestBankKeeper) {
+				// setup transfer channel
+				s.SetupTransferChannel()
+
+				// register a consumer reward denom
+				params := keeper.GetConsumerParams(ctx)
+				params.RewardDenoms = []string{sdk.DefaultBondDenom}
+				keeper.SetParams(ctx, params)
+
+				// send coins to the pool which is used for collect reward distributions to be sent to the provider
+				err := bankKeeper.SendCoinsFromAccountToModule(
+					ctx,
+					s.consumerChain.SenderAccount.GetAddress(),
+					consumertypes.ConsumerToSendToProviderName,
+					sdk.NewCoins(sdk.NewCoin(sdk.DefaultBondDenom, sdk.NewInt(100))),
+				)
+				s.Require().NoError(err)
+			},
+			expError:       false,
+			tokenTransfers: 1,
+		},
+		{
+			name: "no transfer channel",
```

Comment:
I guess this has to do with when `SendRewardsToProvider` returns an error but reading this test's name I would've expected to see `expError: true` because with "no transfer channel" we cannot really send rewards (?) Similarly for the tests below.

Reply:
Yeah, but SendRewardsToProvider returns nil when there is no transfer channel. That's the reason we have `tokenTransfers`